### PR TITLE
Service to application 2

### PR DESCRIPTION
**Pull request by wallyworld**
The second last big instalment in the current work to rename service to application.
This is like unpicking a knitted jumper; you pull on a thread and it gets messy quickly. The intent here is to cover most user visible changes. There are still a lot of comments, internal method names, variable names etc that need to be changed at some point. 

TODO next branch - state methods, db collections, error messages, migration structs

Tested with LXD provider.

Key changes:

api facades renamed, "Service" -> "Application"; "ServiceScaler" -> "ApplicationScaler"
params structs renamed
api methods renamed
api package names
cli args and help text
various go files
facade versions incremented where needed
status yaml
tabular status
many public method names

### CODE REVIEW

File: api/action/client.go [hunk 1]
```diff
@@ -91,20 +91,20 @@ func (c *Client) Cancel(arg params.Actions) (params.ActionResults, error) {
 	return results, err
 }
 
-// servicesCharmActions is a batched query for the charm.Actions for a slice
+// applicationsCharmsActions is a batched query for the charm.Actions for a slice
```
Comment (axw) [inline api/action/client.go]: s/applicationsCharmsActions/applicationsCharmActions/
(stray 's')

File: api/action/client.go [hunk 2]
```diff
@@ -91,20 +91,20 @@ func (c *Client) Cancel(arg params.Actions) (params.ActionResults, error) {
...
-	results := params.ServicesCharmActionsResults{}
-	err := c.facade.FacadeCall("ServicesCharmActions", arg, &results)
+func (c *Client) applicationsCharmsActions(arg params.Entities) (params.ApplicationsCharmActionsResults, error) {
+	results := params.ApplicationsCharmActionsResults{}
+	err := c.facade.FacadeCall("ApplicationsCharmsActions", arg, &results)
 	return results, err
 }
 
-// ServiceCharmActions is a single query which uses ServicesCharmActions to
+// ApplicationCharmActions is a single query which uses ApplicationsCharmsActions to
```
Comment (axw) [inline api/action/client.go]: ditto, and in a couple of other places

**Comment by axw** [comment]
Please s/a application/an application/ over the code base.

File: api/action/client.go (hunk 2 above)
Comment (wallyworld) [inline api/action/client.go]: fixed

File: api/action/client.go (hunk 1 above)
Comment (wallyworld) [inline api/action/client.go]: fixed

**Comment by axw** [comment]
Once "a applicaton" is fixed, LGTM.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**